python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -669,13 +763,94 @@ def stream(
                 def my_dbt_assets(context, dbt: DbtCliResource):
                     yield from dbt.cli(["run"], context=context).stream()
         """
-        for event in self.stream_raw_events():
-            yield from event.to_default_asset_events(
-                manifest=self.manifest,
-                dagster_dbt_translator=self.dagster_dbt_translator,
-                context=self.context,
-                target_path=self.target_path,
-            )
+        # We keep a list of emitted Dagster events and pending futures which augment
+        # emitted events with additional metadata. This ensures we can yield events in the order
+        # they are emitted by dbt.
+        output_events_and_futures: List[Union[Future, DbtDagsterEventType]] = []
+
+        with pushd(str(self.project_dir)):
```

Comment:
Could you explain the `pushd` here?

Reply:
needed to get the duckdb adapter to work correctly - it tries to access the db file assuming the working directory is the project dir - seems like the DBT cli does this in its process, but we need to do the same if we want to use the adapter ourselves